mitodl/open-discussions
# Profile image uploader

#### What are the relevant tickets?
Closes #682

#### What's this PR do?
- Adds the `ProfileImage` container and `ProfileImageUploader` component.
- Adds a rudimentary `ProfilePage` (with url) to be fleshed out in another PR.  Right now it just includes a `ProfileImage` container for testing.  No links to the `ProfilePage` have been introduced yet.
- Removes `SETTINGS.profile_complete` and `SETTINGS.profile_image_small` since they are not dynamic, replaces them with `SETTINGS.profile_ui_editable`
- Modifies the `UserMenu` component to have a `profile` prop so the profile image at top right is automatically refreshed when changes are made to the profile.

#### How should this be manually tested?
- Go to `/profiles/<your_username>`, there will be a rudimentary page with just a `ProfileImage` container on the page.  
- Click the edit button, you should get a dialog with just a 'Cancel' button and a gray box for image files (click to select or drag a file into the box).
- Try dragging a non-image file into the box.  It should not be accepted and an error message should appear just above the box.
- Drag or select a real image.  A 'Save' button should appear, and you should be able to crop the image. 
- Click the 'Save' button.  The dialog should close, and your profile photo should be updated both in the `ProfileImage` component and in the `UserMenu` (top right).
- Click the edit icon again for the image.  The image upload dialog should appear again, with no image selected.
- Go to `/profiles/<some_other_existing_username>`, the `ProfileImage` container on the page should not have an edit button.  

@roberthouse54 a screenshot:
![screen shot 2018-06-15 at 10 56 05 am](https://user-images.githubusercontent.com/187676/41474742-ce4d5f4a-708a-11e8-881b-3503bc0c5992.png)
